src/main/java/thePackmaster/powers/rimworldpack/TradeBeaconPower.java

```diff
@@ -0,0 +1,70 @@
+package thePackmaster.powers.rimworldpack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import thePackmaster.patches.DiscardHookPatch;
+import thePackmaster.powers.AbstractPackmasterPower;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+import static thePackmaster.util.Wiz.adp;
+
+public class TradeBeaconPower extends AbstractPackmasterPower implements DiscardHookPatch.OnDiscardThing{
+    public static final String POWER_ID = makeID(TradeBeaconPower.class.getSimpleName());
+    public static final String NAME = CardCrawlGame.languagePack.getPowerStrings(POWER_ID).NAME;
+    public static final String[] DESCRIPTIONS = CardCrawlGame.languagePack.getPowerStrings(POWER_ID).DESCRIPTIONS;
+
+    boolean traded, postStartDraw;
+
+    public TradeBeaconPower(AbstractCreature owner, int amount) {
+        super(POWER_ID, NAME, PowerType.BUFF, false, owner, amount);
+        traded = false;
+        postStartDraw = true;
+    }
+
+    @Override
+    public void atStartOfTurn() {
+        traded = false;
+        postStartDraw = false;
+    }
+
+    @Override
+    public void atStartOfTurnPostDraw() {
```

Comment:
Per what we discussed in Discord, to make this consistently happen after any other effects that use this hook (most notably powers like DrawCardNextTurn, Brutality, and Tools of the Trade), this should either use priority or a doubly nested action to ensure it happens last.